server/images.go

```diff
@@ -1179,9 +1192,55 @@ func makeRequest(method, url string, headers map[string]string, body io.Reader,
 		return nil, err
 	}
 
+	// if the request is unauthenticated, try to authenticate and make the request again
+	if resp.StatusCode == http.StatusUnauthorized {
```

Comment:
If we get a 401, look for the Www-Authenticate header, parse it, get a JWT and try the call again.

NOTE: since this is a recursive function there probably should be a recursion limit in here, and we should probably abort if the auth string is empty.